Team_Roster/app/routes.py

```diff
@@ -1,27 +1,128 @@
-from flask import request
-from app import app
 from pages.findTeam import ShowFindTeam
 from pages.roster import ShowRoster
 from pages.depthChart import ShowDepthChart
 from pages.immaculateGrid import ShowImmaculateGrid
 
+from flask import flash, redirect, url_for, render_template, request, jsonify
+from flask_login import current_user, login_user, logout_user, login_required
+from urllib.parse import urlsplit
+from app import app, db
+from app.forms import LoginForm, RegistrationForm
+from app.models import User
+import sqlalchemy as sa
+from werkzeug.security import generate_password_hash
+
+@app.route('/', methods=['GET', 'POST'])
+@app.route('/index', methods=['GET', 'POST'])
+@login_required
+def index():
+    if current_user.is_admin:
+        return redirect(url_for('users'))
+    return ShowFindTeam()
+
+@app.route('/users', methods=['GET'])
+@login_required
+def users():
+    if not current_user.is_admin:
+        return redirect(url_for('index'))
+
+    users = User.query.filter(User.username != 'admin').all()
+    return render_template('users.html', users=users)
+
+@app.route('/toggle_user/<int:user_id>', methods=['POST'])
+@login_required
+def toggle_user(user_id):
+    if not current_user.is_admin:
+        return redirect(url_for('index'))
+
+    user = User.query.get_or_404(user_id)
+    user.is_active = not user.is_active
+    db.session.commit()
+    status = "activated" if user.is_active else "deactivated"
+    flash(f"User {user.username} has been {status}.")
+    return redirect(url_for('users'))
 
-@app.route('/',methods=['GET', 'POST'])
-@app.route('/index',methods=['GET', 'POST'])
 @app.route('/findTeam', methods=['GET', 'POST'])
+@login_required
 def findTeam():
     return ShowFindTeam()
 
-@app.route('/<teamId>/roster',methods=['GET'])
+@app.route('/<teamId>/roster', methods=['GET'])
+@login_required
 def roster(teamId):
-    year =request.args.get("year")
-    return ShowRoster(teamId,year)
+    year = request.args.get("year")
+    return ShowRoster(teamId, year)
 
 @app.route('/<teamId>/depthChart', methods=['GET'])
+@login_required
 def depthChart(teamId):
     year = request.args.get("year")
     return ShowDepthChart(teamId, year)
 
 @app.route('/immaculateGrid', methods=['GET','POST'])
 def immaculateGrid():
-    return ShowImmaculateGrid()
\ No newline at end of file
+    return ShowImmaculateGrid()
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if current_user.is_authenticated:
```

Comment:
If you get the chance, it would be good to make  separate login and register functions in their own files to be consistent with the rest of the app.